src/services/analytics/PostHogClient.ts

```diff
@@ -0,0 +1,42 @@
+import { PostHog } from "posthog-node"
+import * as vscode from "vscode"
+
+class PostHogClient {
+	private static instance: PostHogClient
+	private client: PostHog
+	private distinctId: string = vscode.env.machineId
+
+	private constructor() {
+		this.client = new PostHog("phc_5WnLHpYyC30Bsb7VSJ6DzcPXZ34JSF08DJLyM7svZ15", {
+			host: "https://us.i.posthog.com",
+			enableExceptionAutocapture: true,
+		})
+	}
+
+	public static getInstance(): PostHogClient {
+		if (!PostHogClient.instance) {
+			PostHogClient.instance = new PostHogClient()
+		}
+		return PostHogClient.instance
+	}
+
+	public optIn(): void {
+		this.client.identify({ distinctId: this.distinctId })
+		this.client.optIn()
+	}
+
+	public optOut(): void {
+		this.client.optOut()
+	}
+
+	public capture(event: { event: string; properties?: any }): void {
```

Comment:
do we know the properties we want to send? Please consider creating type if we know then. This can eliminate random values being added with out explicitly define them.

Reply:
We don't know. The idea is that engineers can pass whatever data is relevant to the features they're working on